assets/js/modules/auth.js

```diff
@@ -0,0 +1,146 @@
+define([
+    'jquery',
+    'sourceModules/module'
+], function($, module) {
+    
+    'use strict';
+
+    /**
+     * @Object default module option values
+     */
+    var defaults = {
+        'storageKey': 'sourcejsUser',
+        'defaultAvatarURL': '/source/assets/i/unknown.gif',
+        'classes': {
+            'controlsWrapper': 'sjs-login',
+            'loginButton': 'sjs-login-button',
+            'avatar': 'sjs-login-avatar'
+        },
+        'labels': {
+            'login': 'Login',
+            'logout': 'Logout'
+        }
+    };
+
+    /**
+     * @module Auth - basic GitHub authorization module.
+     *
+     * @constructor
+     *
+     * @param [Object] config - auth inline configuration set of options
+     *
+     * @param [Object] config.target - jquery domElement which goes to be auth controlls container
+     *
+     * @param [Object] config.options - options set, which allows to define component configuration.
+     */
+
+    function Auth(config) {
+        var _this = this;
+
+        this.conf = $.extend(true, {},
+            defaults,
+            config.options
+        );
+
+        this.target = config.target || $(this.conf.classes.controlsWrapper);
+        this.popup;
+
+        $(function() {
+            _this.init();
+        });
+    }
+
+
+    Auth.prototype = module.createInstance();
+    Auth.prototype.constructor = Auth;
+
+    Auth.renderers = {
```

Comment:
Why do you leave renderers private? It could be useful to override them in some cases.